src/lib/beacon.ts

```diff
@@ -0,0 +1,42 @@
+import { TestConfiguration } from "../@types"
+
+/**
+ * Beacon.State represents the possible values of the Beacon.Data.state flag.
+ */
+export enum State {
+    /**
+     * The test completed successfully.
+     */
+    Success,
+    /**
+     * The test failed.
+     */
+    Failure,
+    /**
+     * Test status is undetermined.
+     */
+    Unknown,
+}
+
+/**
+ * Beacon.Data represents test data to be sent back according to provider
+ * specifications.
+ */
+export interface Data {
+    /**
+     * The result state of the associated {@link Test}
+     */
+    state: State
+    /**
+     * The configuration of the associated {@link Test}
+     */
+    testConfig: TestConfiguration
+    /**
+     * An object containing provider-defined test data to be beaconed.
+     */
+    data?: {
+        /* eslint-disable @typescript-eslint/no-explicit-any */
+        [key: string]: any
```

Comment:
This is one I've been struggling with a bit, because `SimpleObject` is limited to primitive types (minus Symbol). I'll poke at it some more today and see if I can find a reasonable compromise.

Reply:
I think I got it. I extended SimpleObject a bit and added a test to show it being JSON serializable.